spec/services/govuk_emails/monitor_spec.rb

```diff
@@ -1,108 +1,181 @@
 require "rails_helper"
 
 RSpec.describe GovukEmails::Monitor do
-  let(:scheduled_mailing) { create(:scheduled_mailing, :processing) }
-  let(:response) { double GovukEmails::Email, status: }
-
   before do
     allow(GovukEmails::Email).to receive(:new).with(scheduled_mailing.govuk_message_id).and_return(response)
     Setting.setting.update(alert_via_sentry: true)
   end
 
+  let(:response) { double GovukEmails::Email, status: }
+
   describe ".call" do
-    subject { described_class.call(scheduled_mailing.id) }
+    subject(:call) { described_class.call(scheduled_mailing.id) }
+
+    shared_examples "non-failure handling job" do
```

Comment:
Would it make sense to describe these as "successful" jobs, or is "non-failure" more descriptive?

Reply:
its for anything which is not the `-failure` ones below, and also when not on production 😬 even it is a failure so 🤷  but "successful" does not really match either. open to suggestions

```
    "waiting", # record has been written, but mail not yet sent by ScheduledMailingsDeliveryJob
    "processing", # ScheduledMailingsDeliveryJob is in the process of sending the mail
    "created", # The mail has been sent to GOVUK Notify, but the status has not yet been queries
    "sending", # GOVUK Notify is still trying to send the mail
    "delivered", # GOVUK Notify has confirmed that the mail has been delivered
    "technical-failure", # GOVUK Notify has reported that the mail cannot  be sent due to a technical failure
    "permanent-failure", # GOVUK Notify has reported a permanent failure (most likely, non-existent email address)
    "temporary-failure", # GOVUK Notify has reported a temporary failure
    "cancelled", # Scheduled mail cancelled because it is no longer needed to be sent.
```